react-native/PlantIdentification.js

```diff
@@ -0,0 +1,190 @@
+            allowsEditing: true,
+            aspect: [4, 3],
+            quality: 1,
+            base64: true
+        });
+        
+        if (!result.cancelled) {
+            setPlantImageUri(result.uri)
+            setPlantImage(result);
+        }
+    };
+    
+    const launchImagePicker = async () => {
+        if(!plantImage){
+            return;
+        }
+        const base64Image = ['data:image/jpeg;base64,' + plantImage.base64];
+
+        fetch('https://api.plant.id/v2/identify', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+                api_key: apiKey,
+                images: base64Image,
+            })
+            })
+            .then((response) => {
+                return response.json();
+            })
+            .then((responseJson) => {
+                setPlantDetails(responseJson.suggestions);
+            })
+            .catch((error) => {
+                console.error(error);
+            });
+      };
+        
+    return (
+    <SafeAreaView style={{ backgroundColor: "#F2FFF2", flex: 1 }}>
+        <ScrollView showsVerticalScrollIndicator={false} >
+        <View style={styles.buttonContainer}>
+            <TouchableOpacity onPress={pickImageCamera}>
+              <FontAwesome5Icon
+                name={"camera"} 
+                size={20} 
+                style={{
+                  marginBottom: 3,
+                  alignSelf: "center",
+                  color: "#325232"
+                }} 
+              />
+              <Text style={styles.iconText}>Kamera</Text>
+            </TouchableOpacity>
+            <TouchableOpacity onPress={pickImageLibrary}>
+              <FontAwesome5Icon
+                name={"image"} 
+                size={20} 
+                style={{
+                  marginBottom: 3,
+                  alignSelf: "center",
+                  color: "#325232"
+                }} 
+              />
+              <Text style={styles.iconText}>Gallerie</Text>
+            </TouchableOpacity>
+        </View>
+        <View style={styles.imageContainer}>
+          {
+            plantImageUri !== '' && <Image
+              source={{ uri: plantImageUri }}
+              style={styles.image}
+            /> 
+          }
+        </View>
+        {plantImage && (
+        <>
+            <TouchableOpacity
+                style={styles.saveButton}
+                onPress={() => launchImagePicker()}
+            >
+                <Text style={styles.saveText}>Identify Plant</Text>
+            </TouchableOpacity>
+            {plantDetails && (
+                plantDetails.map((value, index) => {
+                    return <View key={index}>
+                        <Text>Scientific name: {value.plant_details.scientific_name}</Text>
+                        <Text>Probability: {value.plant_details.probability.round(2) * 100 + '%'}</Text>
```

Comment:
This line seems to be invalid, probability is not under plant_details. Also I'm not sure if you can round like this.

> Uncaught TypeError: Cannot read properties of undefined (reading 'round')

>